dataset/src/main/scala/frameless/RecordEncoder.scala

```diff
@@ -169,6 +171,9 @@ class RecordEncoder[F, G <: HList, H <: HList]
         field.encoder.fromCatalyst(fieldPath)
       }
 
-      NewInstance(classTag.runtimeClass, newInstanceExprs.value.from(exprs), jvmRepr, propagateNull = true)
+      val nullExpr = Literal.create(null, jvmRepr)
+      val newArgs = newInstanceExprs.value.from(exprs)
+      val newExpr = NewInstance(classTag.runtimeClass, newArgs, jvmRepr, propagateNull = true)
```

Comment:
I would change `propagateNull` to `false` as [it is how Spark works now](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala#L404), but it seems to cause a lot of other problems now.